src/test/regress/sql/multi_mx_function_call_delegation.sql

```diff
@@ -1,69 +1,147 @@
 -- Test passing off function call to mx workers
--- Create worker-local tables to test procedure calls were routed
+-- Create worker-local tables to test function calls were routed
 CREATE SCHEMA multi_mx_function_call_delegation;
 SET search_path TO multi_mx_function_call_delegation;
-SET citus.replication_model TO 'streaming';
-SET citus.shard_replication_factor TO 1;
 
-create table mx_call_dist_table(id int, val int);
-select create_distributed_table('mx_call_dist_table', 'id');
+SET citus.shard_replication_factor TO 2;
+SET citus.replication_model TO 'statement';
 
-CREATE FUNCTION mx_call_func(INOUT x int, INOUT y int) RETURNS record LANGUAGE plpgsql AS $$
-BEGIN
-    y := (select groupid from pg_dist_local_group) + x + y;
-END;$$;
+-- This table requires specific settings, create before getting into things
+create table mx_call_dist_table_replica(id int, val int);
+select create_distributed_table('mx_call_dist_table_replica', 'id');
+insert into mx_call_dist_table_replica values (9,1),(8,2),(7,3),(6,4),(5,5);
 
-CREATE FUNCTION mx_call_func_raise(x int) RETURNS void LANGUAGE plpgsql AS $$
-BEGIN
-    RAISE WARNING 'warning';
-END;$$;
+SET citus.shard_replication_factor TO 1;
+SET citus.replication_model TO 'streaming';
 
-CREATE FUNCTION mx_call_add(int, int) RETURNS int
-    AS 'select $1 + $2;' LANGUAGE SQL IMMUTABLE;
+--
+-- Utility UDFs
+--
 
-CREATE FUNCTION mx_call_dist_object(procname text, tablerelid regclass, argument_index int) RETURNS void LANGUAGE plpgsql AS $$
+-- 1. Marks the given function as colocated with the given table.
+-- 2. Marks the argument index with which we route the function.
+CREATE FUNCTION colocate_func_with_table(procname text, tablerelid regclass, argument_index int)
```

Comment:
Maybe move this as `PROCEDURE` to `multi_test_helpers.sql`, since it is used in more than one test files. Similar to https://github.com/citusdata/citus/pull/3032/files